lib/cfg-tree.c

```diff
@@ -1359,6 +1403,13 @@ cfg_tree_compile(CfgTree *self)
           return FALSE;
         }
     }
+
+  if (self->check_zero_source_count && !cfg_tree_check_empty_pipelines(self))
```

Comment:
I think the validation step could be a separate function that the main program calls and the test programs don't. This way test programs don't need to opt out of validation by setting a boolean to false.

I would envision similar methods, like cfg_tree_optimize() that could remove no-op logpipe instances from the graph.

Reply:
I'll check this.